### PR TITLE
Make the tests run by transforming warnings into errors

### PR DESCRIPTION
**Pull request by mohitreddy1996**
Introduce pytest.ini and introduce configuration to filter and transform warnings into errors.

Few exceptions are documented in the same file. We filter out the specific warning using regex and its type instead of ignoring a generic type of warning which might lead to us missing out on something newly introduced in the flax codebase. Notably causing due to:
- Importing tensorflow.compat.v2.io import gfile in training/checkpoints.py
- XLA level Warning.
- RuntimeWarning due to scipy compiled against an older numpy than installed.

@avital PTAL.

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[pytest]
+# RuntimeWarning is due to scipy (or another package) that was compiled against an older numpy than is installed.
+# UserWarning is due to statement: tensorflow.compat.v2.io import gfile
+# DeprecationWarning is due to statement: tensorflow.compat.v2.io import gfile
```
Comment (avital) [inline pytest.ini:4]: Should we be actually importing gfile in a differnet way?

**Comment by avital** [comment]
Thanks, this is a clear improvement! Should we file separate GitHub tickets to investigate why the other deprecation warnings happen and see if we can eliminate them altogether?

File: pytest.ini (hunk 1 above)
Comment (avital) [reply pytest.ini:4]: (to avoid this warning)

Comment (mohitreddy1996) [reply pytest.ini:4]: Good suggestion. I didn't try it, but let me give it a shot. Thanks!

**Comment by mohitreddy1996** [comment]
> Thanks, this is a clear improvement! Should we file separate GitHub tickets to investigate why the other deprecation warnings happen and see if we can eliminate them altogether?

Yes, I will go ahead and create them.

**Comment by avital** [comment]
Thanks @mohitreddy1996!

**Pull request merged**